superset/migrations/versions/8b841273bec3_sql_lab_models_database_constraint_updates.py

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""sql_lab_models_database_constraint_updates
+
+Revision ID: 8b841273bec3
+Revises: 2ed890b36b94
+Create Date: 2022-03-16 21:07:48.768425
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "8b841273bec3"
+down_revision = "2ed890b36b94"
+
+import sqlalchemy as sa
+from alembic import op
+
+from superset.utils.core import generic_find_fk_constraint_name
+
+
+def upgrade():
+    bind = op.get_bind()
+    insp = sa.engine.reflection.Inspector.from_engine(bind)
+
+    with op.batch_alter_table("tab_state") as batch_op:
+        batch_op.drop_constraint(
+            generic_find_fk_constraint_name("tab_state", {"id"}, "dbs", insp),
+            type_="foreignkey",
+        )
+
+        batch_op.drop_constraint(
+            generic_find_fk_constraint_name("tab_state", {"client_id"}, "query", insp),
+            type_="foreignkey",
+        )
+
+        batch_op.create_foreign_key(
+            "tab_state_database_id_fkey",
+            "dbs",
+            ["database_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+
+        batch_op.create_foreign_key(
+            "tab_state_latest_query_id_fkey",
+            "query",
+            ["latest_query_id"],
+            ["client_id"],
+            ondelete="SET NULL",
+        )
+
+    with op.batch_alter_table("table_schema") as batch_op:
+        batch_op.drop_constraint(
```

Comment:
for the sqlite errors here, it's because sqlite doesn't support foreign key deletion and actually these foreign key lookups are returning None. So you could either check that the foreign key exists before deleting it or only run these foreign key deletions if the db is not sqlite like:
```
if not isinstance(bind.dialect, SQLiteDialect):
```

Reply:
so, in order to support sqlite, can we do that? Or do we need to rename this table, create the new one with the proper constraints and then delete the old one?

Reply:
You can just not run the drop_constraint if the dialect is sqlite or if the constraint isn't found. I prefer the latter b/c it's more specific and descriptive, but either would work. There are a lot of instance of checking for SQLiteDialect in the codebase already. If you go with the latter, though, it would be something like this: 
```
table_schema_id_constraint = generic_find_fk_constraint_name("table_schema", {"id"}, "dbs", insp)
if table_schema_id_constraint:
    batch_op.drop_constraint(table_schema_id_constraint, type_="foreignkey")
```

Reply:
The issue in this case is that I'm dropping the constraint here to add it again, with a different setup.
Will it work if the field has more than one constraint? seems odd to me, but if it does, then will do that.

Reply:
Yeah, Sqlite has foreign keys turned off by default, so if you do the lookup and it's not found, then you should be fine.